File: backend-go/internal/metrics/metrics.go
package metrics

import (
	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promauto"
)

var (
	AuditsTotal = promauto.NewCounterVec(
		prometheus.CounterOpts{
			Name: "trustagent_audits_total",
			Help: "Total number of audits processed",
		},
		[]string{"status"},
	)

	HallucinationsDetected = promauto.NewCounter(
		prometheus.CounterOpts{
			Name: "trustagent_hallucinations_detected_total",
			Help: "Total number of hallucinations detected",
		},
	)

	AuditDuration = promauto.NewHistogram(
		prometheus.HistogramOpts{
			Name:    "trustagent_audit_duration_seconds",
			Help:    "Audit processing duration in seconds",
			Buckets: []float64{0.5, 1, 2, 5, 10, 30, 60},
		},
	)

	FaithfulnessScore = promauto.NewHistogram(
		prometheus.HistogramOpts{
			Name:    "trustagent_faithfulness_score",
			Help:    "Distribution of faithfulness scores",
			Buckets: []float64{0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0},
		},
	)

	ActiveAudits = promauto.NewGauge(
		prometheus.GaugeOpts{
			Name: "trustagent_active_audits",
			Help: "Number of currently processing audits",
		},
	)

	WebSocketClients = promauto.NewGauge(
		prometheus.GaugeOpts{
			Name: "trustagent_websocket_clients",
			Help: "Number of connected WebSocket clients",
		},
	)

	ClaimsTotal = promauto.NewCounterVec(
		prometheus.CounterOpts{
			Name: "trustagent_claims_total",
			Help: "Total claims by verification status",
		},
		[]string{"status"},
	)
)

File: backend-python/src/truthtable/metrics.py
"""
Prometheus metrics for the TrustAgent audit engine.
"""

from prometheus_client import Counter, Histogram, Gauge

AUDITS_TOTAL = Counter(
    "truthtable_audits_total",
    "Total number of audits processed",
    ["status"],
)

AUDIT_DURATION = Histogram(
    "truthtable_audit_duration_seconds",
    "Audit processing duration in seconds",
    buckets=[0.5, 1, 2, 5, 10, 30, 60],
)

FAITHFULNESS_SCORE = Histogram(
    "truthtable_faithfulness_score",
    "Distribution of faithfulness scores",
    buckets=[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0],
)

HALLUCINATIONS_DETECTED = Counter(
    "truthtable_hallucinations_detected_total",
    "Total number of hallucinations detected",
)

ACTIVE_AUDITS = Gauge(
    "truthtable_active_audits",
    "Number of currently processing audits",
)

CLAIMS_TOTAL = Counter(
    "truthtable_claims_total",
    "Total claims by verification status",
    ["status"],
)
